File: solution/java/week05-complexity/exercise2/src/test/java/command/FakeOutputAdapter.java
package command;

import io.vavr.collection.List;

public class FakeOutputAdapter {
    List<String> allOutputs = List.empty();

    public String getAllOutputs(){
        return allOutputs.mkString("\n");
    }

    public void sendOut(String message) {
        allOutputs = allOutputs.append(message);
    }
}

File: solution/kotlin/week05-exercise2/src/test/kotlin/FakeOutputAdapter.kt
package command

import java.lang.System.lineSeparator

class FakeOutputAdapter {
    private val allOutputs: MutableList<String> = mutableListOf()

    fun getAllOutputs(): String = allOutputs.joinToString(lineSeparator())

    fun sendOut(message: String) {
        allOutputs.add(message)
    }
}
